contracts/base/RegistryAdapter.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import { IERC7484 } from "../interfaces/IERC7484.sol";
+
+/**
+ * IERC7484 Registry adapter.
+ * this feature is opt-in. The smart account owner can choose to use the registry and which
+ * attesters to trust
+ */
+abstract contract RegistryAdapter {
+    event ERC7484RegistryConfigured(IERC7484 indexed registry);
+
+    IERC7484 registry;
+
+    modifier withRegistry(address module, uint256 moduleType) {
+        _checkRegistry(module, moduleType);
+        _;
+    }
+
+    /**
+     * Check on ERC7484 Registry, if suffcient attestations were made
+     * This will revert, if not succicient valid attestations are on the registry
+     */
+    function _checkRegistry(address module, uint256 moduleType) internal view {
+        IERC7484 _registry = registry;
+        if (address(_registry) != address(0)) {
+            // this will revert if attestations / threshold are not met
+            _registry.check(module, moduleType);
```

Comment:
if we do this everytime through modifier
_checkRegistry

that means, we will always call check() function 
and SA should have called trustAttestors before that?

Reply:
when could these be used?
function checkN(address module, address[] calldata attesters, uint256 threshold) external view;
function checkN(address module, uint256 moduleType, address[] calldata attesters, uint256 threshold) external view;

if we enshrine this current code

Reply:
yes thats right, the smart account should've set their trusted attesters before

Reply:
imo it makes more sense to store attesters in the registry since they can then also eg be used by a session key module to check sub-modules on the registry